scripts/sct_image.py

```diff
@@ -195,14 +193,16 @@ def main(args=None):
         im_out = [concat_data(fname_in, dim)]  # TODO: adapt to fname_in
 
     elif arguments.copy_header is not None:
+        if fname_out is None:
+            raise ValueError("Need to specify output image with -o!")
         im_in = Image(fname_in[0])
-        im_dest = Image(arguments.copy_header)
+        im_dest = Image(fname_out)
         im_dest_new = im_in.copy()
         im_dest_new.data = im_dest.data.copy()
         # im_dest.header = im_in.header
         im_dest_new.absolutepath = im_dest.absolutepath
         im_out = [im_dest_new]
-        fname_out = arguments.copy_header
+        fname_out = fname_out
```

Comment:
i'm not a big fan of this approach, because this will overwrite the destination image (the one you targeted as `-o`). In a typical scenario, we have src.nii, dest.nii, and we would like to create dest_with_src_header.nii (not always overwrite dest.nii).

Reply:
Woops this is a mistake from when I amended my previous commit. This line shouldn't exist. Fixed in bf2b00a

Reply:
So basically the behavior is that if the user wants to use `copy_header`, he must provide the output image (like before), but via `-o`.

Reply:
> So basically the behavior is that if the user wants to use `-copy-header`, he must provide the output image (like before), but via `-o` instead of via a value to `-copy-header`.

Reply:
i have a small problem with that approach. In some instances, users wouldn't like to have their images being overwritten: this is what would happen here (ie: the file specified by `-o` would be overwritten). 

Alternatively, it would be great to give the option to duplicate the destination file and output it as another file. So if we go with that approach, `-copy-header` would keep the previous behaviour (ie: point to the destination file), and if `-o` exists, the destination file will be duplicated, named as `-o` and the header will be changed in that duplicated file. Does that make sense?

Reply:
Ah ok I get it. Sorry seems I misunderstood what was required. Will adjust.

Reply:
fixed 4cb0708

Reply:
I was thinking of using -copy-header if -o is not specified, but your enforcement approach (ie -o mandatory) is probably the safest